### DISCUSSION TITLE
Is there any way to invalidate fetch cache between builds?

### DISCUSSION THREAD

**Discussion by jaisharx** [vercel/next.js#66623] [Help]
### Summary

I have an endpoint for translation data, which is fetched per route level when doing ```next build```. If I put cache: 'no-cache' in the fetch options, then it makes too many requests to the service, and if I put cache: 'force-cache', then the cache is shared between builds on Vercel, so if I push another commit there is old translation data being used.

I'm wondering if there is a way to invalidate the cache between new builds?

**Comment by SouthLink** [comment]
If you need fine-grained cache control you may need to [customize cache](https://nextjs.org/docs/app/api-reference/next-config-js/incrementalCacheHandlerPath)，use a script to invalidate the cache you need when pushing a new version

**Reply by jaisharx** [reply]
Thanks, I have created a custom cache handler as the docs says here: https://nextjs.org/docs/app/building-your-application/deploying#configuring-caching

But they don't show how to consume this cache? Do I use the unstable_cache from 'next/cache' or what's the API to consume the get/set/revalidateTag methods in a script?

**Reply by SouthLink** [reply]
> Thanks, I have created a custom cache handler as the docs says here: https://nextjs.org/docs/app/building-your-application/deploying#configuring-caching
> 
> But they don't show how to consume this cache? Do I use the unstable_cache from 'next/cache' or what's the API to consume the get/set/revalidateTag methods in a script?

You can refer to the official cache-handler example : https://github.com/vercel/next.js/tree/canary/examples/cache-handler-redis

**Reply by jaisharx** [reply]
Thanks for linking this example, but it's not helpful for what I want to achieve. I want a way to invalidate cache between different builds, what the example does is attach the local cache to a Redis instance for remote persistence.

**Reply by SouthLink** [reply]
> Thanks for linking this example, but it's not helpful for what I want to achieve. I want a way to invalidate cache between different builds, what the example does is attach the local cache to a Redis instance for remote persistence.

What I mean is that you can manage caches centrally, for example by adding prefix to the redis key to manage the individual build caches, and partition them, and then use a script to clean these caches uniformly when starting the project
